crates/rome_js_analyze/src/analyzers/nursery/no_fallthrough_switch_clause.rs

```diff
@@ -47,36 +59,142 @@ declare_rule! {
 }
 
 impl Rule for NoFallthroughSwitchClause {
-    type Query = Ast<JsSwitchStatement>;
-    type State = AnyJsSwitchClause;
+    type Query = ControlFlowGraph;
+    type State = TextRange;
     type Signals = Vec<Self::State>;
     type Options = ();
 
     fn run(ctx: &RuleContext<Self>) -> Self::Signals {
-        let query = ctx.query();
-        let mut signals: Self::Signals = Vec::new();
-        let mut cases = query.cases().into_iter().peekable();
-
-        while let Some(any_case) = cases.next() {
-            let is_last = cases.peek().is_none();
-
-            if is_last {
+        let cfg = ctx.query();
+        let mut fallthrough: Vec<TextRange> = vec![];
+        // Return early if the graph doesn't contain any switch statements.
+        // This avoids to allocate some memory.
+        if !has_switch_statement(&cfg.node) {
+            return fallthrough;
+        }
+        // block to process.
+        let mut block_stack = Vec::new();
+        let mut visited_blocks = FxHashSet::default();
+        block_stack.push(0u32);
+        visited_blocks.insert(0u32);
```

Comment:
What does this insertion do?

Reply:
It marks the first block as visited (every time we insert a block id in the stack we have to register that it was visited).